lib/tasks/grade.rake

```diff
@@ -128,37 +132,60 @@ namespace :grade do
     end
     puts "Grade token has been reset successfully."
   end
+end
+
+def sync_specs_with_source(full_reponame, remote_sha, repo_url)
```

Comment:
Perhaps related to some of the changes on this method. As a student, I deleted a one spec, then _staged_ (but did not _commit_) the file. I then ran `grade`, and I seemed to have tricked `Grades` into thinking there was one less spec than before. I'm not sure if this was actually behavior that occurred prior to this PR as it only just occurred to me to test this somewhat edge-case (we never describe / use "staging" for students and just have them select "Always" the first time they commit without staging)..

--- 
Student codespace:

![image](https://github.com/user-attachments/assets/6fc4e6a5-f3df-467c-951f-0f1853b88fc2)

---
Tricked grades:

![image](https://github.com/user-attachments/assets/9833c325-19d9-4604-8c4f-c380cc76d35f)

Reply:
And just want to confirm that if I actually commit the changes (or just don't stage in the first place), it does the right thing and downloads the "official" spec folder again.

Reply:
Oh good catch. I hadn't tested the case where the SHA of the local spec folder matches the remote _and_ a spec file change has been staged. This was a problem previously but can be handled if grade runner unstages staged files from the spec folder before comparing SHA's.